src/main/groovy/com/pszymczyk/consul/ConsulProcess.groovy

```diff
@@ -1,22 +1,37 @@
 package com.pszymczyk.consul
 
+import com.pszymczyk.consul.infrastructure.SimpleConsulClient
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory;
 
 import java.nio.file.Path;
 
-public class ConsulProcess implements AutoCloseable {
+class ConsulProcess implements AutoCloseable {
 
     private static final Logger logger = LoggerFactory.getLogger(ConsulProcess.class);
 
-    Path dataDir;
-    int httpPort;
-    Process process;
+    private Path dataDir
+    private int httpPort
+    private Process process
+    private SimpleConsulClient simpleConsulClient
+
+    ConsulProcess(Path dataDir, int httpPort, Process process) {
+        this.dataDir = dataDir
+        this.httpPort = httpPort
+        this.process = process
+        this.simpleConsulClient = new SimpleConsulClient(httpPort: httpPort)
+    }
+    /**
+     * deregister all services except consul
+     */
+    void reset() {
+        simpleConsulClient.getServicesIds().each { it -> simpleConsulClient.deregister(it) }
```

Comment:
No it isn't. 
To list all values use `curl http://localhost:8500/v1/kv/?recurse`
To delete all values use `curl -X DELETE http://localhost:8500/v1/kv/?recurse`